pkg/kubelet/kubelet_pods.go

```diff
@@ -1728,8 +1743,11 @@ func hasHostVolume(pod *v1.Pod) bool {
 	return false
 }
 
-// hasHostNamespace returns true if hostIPC, hostNetwork, or hostPID are set to true.
+// hasHostNamespace returns true if hostIPC, hostNetwork, or hostPID are set to true or userNamespaceRemapping is set to false.
 func hasHostNamespace(pod *v1.Pod) bool {
+	if pod.Spec.HostUser != nil {
+		return pod.Spec.HostIPC || pod.Spec.HostNetwork || pod.Spec.HostPID || *pod.Spec.HostUser
+	}
 	if pod.Spec.SecurityContext == nil {
 		return false
 	}
```

Comment:
could add HostUser to the existing list with `... || (pod.Spec.HostUser != nil && *pod.Spec.HostUser)`

Reply:
sure